src/python/grpcio_health_checking/grpc_health/v1/health.py

```diff
@@ -23,30 +23,92 @@
 SERVICE_NAME = _health_pb2.DESCRIPTOR.services_by_name['Health'].full_name
 
 
+class _Watcher():
+
+    def __init__(self):
+        self._condition = threading.Condition()
+        self._responses = list()
+        self._open = True
+
+    def __iter__(self):
+        return self
+
+    def _next(self):
+        with self._condition:
+            while not self._responses and self._open:
```

Comment:
Optional: In current implementation, only newly added status or 'close' being called will awake this thread. It is safe to use `if` here.

Reply:
Acknowledged